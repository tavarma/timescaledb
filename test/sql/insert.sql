\ir include/insert_two_partitions.sql

\d+ "_timescaledb_internal".*
SELECT * FROM _timescaledb_catalog.chunk;

SELECT * FROM "two_Partitions" ORDER BY "timeCustom", device_id;
SELECT * FROM ONLY "two_Partitions";

CREATE TABLE error_test(time timestamp, temp float8, device text NOT NULL);
SELECT create_hypertable('error_test', 'time', 'device', 2);

\set QUIET off
INSERT INTO error_test VALUES ('Mon Mar 20 09:18:20.1 2017', 21.3, 'dev1');
\set ON_ERROR_STOP 0
-- generate insert error 
INSERT INTO error_test VALUES ('Mon Mar 20 09:18:22.3 2017', 21.1, NULL);
\set ON_ERROR_STOP 1
INSERT INTO error_test VALUES ('Mon Mar 20 09:18:25.7 2017', 22.4, 'dev2');
\set QUIET on
SELECT * FROM error_test;

--test character(9) partition keys since there were issues with padding causing partitioning errors
CREATE TABLE tick_character (
    symbol      character(9) NOT NULL,
    mid       REAL NOT NULL,
    spread      REAL NOT NULL,
    time        TIMESTAMPTZ       NOT NULL
);

SELECT create_hypertable ('tick_character', 'time', 'symbol', 2);
INSERT INTO tick_character ( symbol, mid, spread, time ) VALUES ( 'GBPJPY', 142.639000, 5.80, 'Mon Mar 20 09:18:22.3 2017') RETURNING time, symbol, mid;
SELECT * FROM tick_character;


CREATE TABLE  many_partitions_test(time timestamp, temp float8, device text NOT NULL);
SELECT create_hypertable('many_partitions_test', 'time', 'device', 1000);
--NOTE: how much slower the first two queries are -- they are creating chunks
INSERT INTO many_partitions_test
    SELECT to_timestamp(ser), ser, ser::text FROM generate_series(1,100) ser;
INSERT INTO many_partitions_test
    SELECT to_timestamp(ser), ser, ser::text FROM generate_series(101,200) ser;
INSERT INTO many_partitions_test
    SELECT to_timestamp(ser), ser, (ser-201)::text FROM generate_series(201,300) ser;

SELECT * FROM  many_partitions_test ORDER BY time DESC LIMIT 2;
SELECT count(*) FROM  many_partitions_test;

CREATE TABLE  date_col_test(time date, temp float8, device text NOT NULL);
SELECT create_hypertable('date_col_test', 'time', 'device', 1000, chunk_time_interval => INTERVAL '1 Day');
INSERT INTO date_col_test
VALUES ('2001-02-01', 98, 'dev1'),
('2001-03-02', 98, 'dev1');

SELECT * FROM date_col_test WHERE time > '2001-01-01';
