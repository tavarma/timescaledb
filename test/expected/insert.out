\ir include/insert_two_partitions.sql
\ir create_single_db.sql
SET client_min_messages = WARNING;
DROP DATABASE IF EXISTS single;
SET client_min_messages = NOTICE;
CREATE DATABASE single;
\c single
CREATE EXTENSION IF NOT EXISTS timescaledb;
SET timescaledb.disable_optimizations = :DISABLE_OPTIMIZATIONS;
\c single
CREATE TABLE PUBLIC."two_Partitions" (
  "timeCustom" BIGINT NOT NULL,
  device_id TEXT NOT NULL,
  series_0 DOUBLE PRECISION NULL,
  series_1 DOUBLE PRECISION NULL,
  series_2 DOUBLE PRECISION NULL,
  series_bool BOOLEAN NULL
);
CREATE INDEX ON PUBLIC."two_Partitions" (device_id, "timeCustom" DESC NULLS LAST) WHERE device_id IS NOT NULL;
CREATE INDEX ON PUBLIC."two_Partitions" ("timeCustom" DESC NULLS LAST, series_0) WHERE series_0 IS NOT NULL;
CREATE INDEX ON PUBLIC."two_Partitions" ("timeCustom" DESC NULLS LAST, series_1)  WHERE series_1 IS NOT NULL;
CREATE INDEX ON PUBLIC."two_Partitions" ("timeCustom" DESC NULLS LAST, series_2) WHERE series_2 IS NOT NULL;
CREATE INDEX ON PUBLIC."two_Partitions" ("timeCustom" DESC NULLS LAST, series_bool) WHERE series_bool IS NOT NULL;
CREATE INDEX ON PUBLIC."two_Partitions" ("timeCustom" DESC NULLS LAST, device_id);
SELECT * FROM create_hypertable('"public"."two_Partitions"'::regclass, 'timeCustom'::name, 'device_id'::name, associated_schema_name=>'_timescaledb_internal'::text, number_partitions => 2, chunk_time_interval=>_timescaledb_internal.interval_to_usec('1 month'));
 create_hypertable 
-------------------
 
(1 row)

\set QUIET off
BEGIN;
BEGIN
\COPY public."two_Partitions" FROM 'data/ds1_dev1_1.tsv' NULL AS '';
COPY 7
COMMIT;
COMMIT
INSERT INTO public."two_Partitions"("timeCustom", device_id, series_0, series_1) VALUES
(1257987600000000000, 'dev1', 1.5, 1),
(1257987600000000000, 'dev1', 1.5, 2),
(1257894000000000000, 'dev2', 1.5, 1),
(1257894002000000000, 'dev1', 2.5, 3);
INSERT 0 4
INSERT INTO "two_Partitions"("timeCustom", device_id, series_0, series_1) VALUES
(1257894000000000000, 'dev2', 1.5, 2);
INSERT 0 1
\set QUIET on
\d+ "_timescaledb_internal".*
Index "_timescaledb_internal.1-two_Partitions_device_id_timeCustom_idx"
   Column   |  Type  |  Definition  | Storage  
------------+--------+--------------+----------
 device_id  | text   | device_id    | extended
 timeCustom | bigint | "timeCustom" | plain
btree, for table "_timescaledb_internal._hyper_1_1_chunk", predicate (device_id IS NOT NULL)

Index "_timescaledb_internal.10-two_Partitions_timeCustom_series_1_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_1   | double precision | series_1     | plain
btree, for table "_timescaledb_internal._hyper_1_2_chunk", predicate (series_1 IS NOT NULL)

Index "_timescaledb_internal.11-two_Partitions_timeCustom_series_2_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_2   | double precision | series_2     | plain
btree, for table "_timescaledb_internal._hyper_1_2_chunk", predicate (series_2 IS NOT NULL)

Index "_timescaledb_internal.12-two_Partitions_timeCustom_series_bool_idx"
   Column    |  Type   |  Definition  | Storage 
-------------+---------+--------------+---------
 timeCustom  | bigint  | "timeCustom" | plain
 series_bool | boolean | series_bool  | plain
btree, for table "_timescaledb_internal._hyper_1_2_chunk", predicate (series_bool IS NOT NULL)

Index "_timescaledb_internal.13-two_Partitions_timeCustom_device_id_idx"
   Column   |  Type  |  Definition  | Storage  
------------+--------+--------------+----------
 timeCustom | bigint | "timeCustom" | plain
 device_id  | text   | device_id    | extended
btree, for table "_timescaledb_internal._hyper_1_2_chunk"

Index "_timescaledb_internal.14-two_Partitions_timeCustom_idx"
   Column   |  Type  |  Definition  | Storage 
------------+--------+--------------+---------
 timeCustom | bigint | "timeCustom" | plain
btree, for table "_timescaledb_internal._hyper_1_2_chunk"

Index "_timescaledb_internal.15-two_Partitions_device_id_timeCustom_idx"
   Column   |  Type  |  Definition  | Storage  
------------+--------+--------------+----------
 device_id  | text   | device_id    | extended
 timeCustom | bigint | "timeCustom" | plain
btree, for table "_timescaledb_internal._hyper_1_3_chunk", predicate (device_id IS NOT NULL)

Index "_timescaledb_internal.16-two_Partitions_timeCustom_series_0_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_0   | double precision | series_0     | plain
btree, for table "_timescaledb_internal._hyper_1_3_chunk", predicate (series_0 IS NOT NULL)

Index "_timescaledb_internal.17-two_Partitions_timeCustom_series_1_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_1   | double precision | series_1     | plain
btree, for table "_timescaledb_internal._hyper_1_3_chunk", predicate (series_1 IS NOT NULL)

Index "_timescaledb_internal.18-two_Partitions_timeCustom_series_2_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_2   | double precision | series_2     | plain
btree, for table "_timescaledb_internal._hyper_1_3_chunk", predicate (series_2 IS NOT NULL)

Index "_timescaledb_internal.19-two_Partitions_timeCustom_series_bool_idx"
   Column    |  Type   |  Definition  | Storage 
-------------+---------+--------------+---------
 timeCustom  | bigint  | "timeCustom" | plain
 series_bool | boolean | series_bool  | plain
btree, for table "_timescaledb_internal._hyper_1_3_chunk", predicate (series_bool IS NOT NULL)

Index "_timescaledb_internal.2-two_Partitions_timeCustom_series_0_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_0   | double precision | series_0     | plain
btree, for table "_timescaledb_internal._hyper_1_1_chunk", predicate (series_0 IS NOT NULL)

Index "_timescaledb_internal.20-two_Partitions_timeCustom_device_id_idx"
   Column   |  Type  |  Definition  | Storage  
------------+--------+--------------+----------
 timeCustom | bigint | "timeCustom" | plain
 device_id  | text   | device_id    | extended
btree, for table "_timescaledb_internal._hyper_1_3_chunk"

Index "_timescaledb_internal.21-two_Partitions_timeCustom_idx"
   Column   |  Type  |  Definition  | Storage 
------------+--------+--------------+---------
 timeCustom | bigint | "timeCustom" | plain
btree, for table "_timescaledb_internal._hyper_1_3_chunk"

Index "_timescaledb_internal.22-two_Partitions_device_id_timeCustom_idx"
   Column   |  Type  |  Definition  | Storage  
------------+--------+--------------+----------
 device_id  | text   | device_id    | extended
 timeCustom | bigint | "timeCustom" | plain
btree, for table "_timescaledb_internal._hyper_1_4_chunk", predicate (device_id IS NOT NULL)

Index "_timescaledb_internal.23-two_Partitions_timeCustom_series_0_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_0   | double precision | series_0     | plain
btree, for table "_timescaledb_internal._hyper_1_4_chunk", predicate (series_0 IS NOT NULL)

Index "_timescaledb_internal.24-two_Partitions_timeCustom_series_1_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_1   | double precision | series_1     | plain
btree, for table "_timescaledb_internal._hyper_1_4_chunk", predicate (series_1 IS NOT NULL)

Index "_timescaledb_internal.25-two_Partitions_timeCustom_series_2_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_2   | double precision | series_2     | plain
btree, for table "_timescaledb_internal._hyper_1_4_chunk", predicate (series_2 IS NOT NULL)

Index "_timescaledb_internal.26-two_Partitions_timeCustom_series_bool_idx"
   Column    |  Type   |  Definition  | Storage 
-------------+---------+--------------+---------
 timeCustom  | bigint  | "timeCustom" | plain
 series_bool | boolean | series_bool  | plain
btree, for table "_timescaledb_internal._hyper_1_4_chunk", predicate (series_bool IS NOT NULL)

Index "_timescaledb_internal.27-two_Partitions_timeCustom_device_id_idx"
   Column   |  Type  |  Definition  | Storage  
------------+--------+--------------+----------
 timeCustom | bigint | "timeCustom" | plain
 device_id  | text   | device_id    | extended
btree, for table "_timescaledb_internal._hyper_1_4_chunk"

Index "_timescaledb_internal.28-two_Partitions_timeCustom_idx"
   Column   |  Type  |  Definition  | Storage 
------------+--------+--------------+---------
 timeCustom | bigint | "timeCustom" | plain
btree, for table "_timescaledb_internal._hyper_1_4_chunk"

Index "_timescaledb_internal.3-two_Partitions_timeCustom_series_1_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_1   | double precision | series_1     | plain
btree, for table "_timescaledb_internal._hyper_1_1_chunk", predicate (series_1 IS NOT NULL)

Index "_timescaledb_internal.4-two_Partitions_timeCustom_series_2_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_2   | double precision | series_2     | plain
btree, for table "_timescaledb_internal._hyper_1_1_chunk", predicate (series_2 IS NOT NULL)

Index "_timescaledb_internal.5-two_Partitions_timeCustom_series_bool_idx"
   Column    |  Type   |  Definition  | Storage 
-------------+---------+--------------+---------
 timeCustom  | bigint  | "timeCustom" | plain
 series_bool | boolean | series_bool  | plain
btree, for table "_timescaledb_internal._hyper_1_1_chunk", predicate (series_bool IS NOT NULL)

Index "_timescaledb_internal.6-two_Partitions_timeCustom_device_id_idx"
   Column   |  Type  |  Definition  | Storage  
------------+--------+--------------+----------
 timeCustom | bigint | "timeCustom" | plain
 device_id  | text   | device_id    | extended
btree, for table "_timescaledb_internal._hyper_1_1_chunk"

Index "_timescaledb_internal.7-two_Partitions_timeCustom_idx"
   Column   |  Type  |  Definition  | Storage 
------------+--------+--------------+---------
 timeCustom | bigint | "timeCustom" | plain
btree, for table "_timescaledb_internal._hyper_1_1_chunk"

Index "_timescaledb_internal.8-two_Partitions_device_id_timeCustom_idx"
   Column   |  Type  |  Definition  | Storage  
------------+--------+--------------+----------
 device_id  | text   | device_id    | extended
 timeCustom | bigint | "timeCustom" | plain
btree, for table "_timescaledb_internal._hyper_1_2_chunk", predicate (device_id IS NOT NULL)

Index "_timescaledb_internal.9-two_Partitions_timeCustom_series_0_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_0   | double precision | series_0     | plain
btree, for table "_timescaledb_internal._hyper_1_2_chunk", predicate (series_0 IS NOT NULL)

                   Table "_timescaledb_internal._hyper_1_1_chunk"
   Column    |       Type       | Modifiers | Storage  | Stats target | Description 
-------------+------------------+-----------+----------+--------------+-------------
 timeCustom  | bigint           | not null  | plain    |              | 
 device_id   | text             | not null  | extended |              | 
 series_0    | double precision |           | plain    |              | 
 series_1    | double precision |           | plain    |              | 
 series_2    | double precision |           | plain    |              | 
 series_bool | boolean          |           | plain    |              | 
Indexes:
    "1-two_Partitions_device_id_timeCustom_idx" btree (device_id, "timeCustom" DESC NULLS LAST) WHERE device_id IS NOT NULL
    "2-two_Partitions_timeCustom_series_0_idx" btree ("timeCustom" DESC NULLS LAST, series_0) WHERE series_0 IS NOT NULL
    "3-two_Partitions_timeCustom_series_1_idx" btree ("timeCustom" DESC NULLS LAST, series_1) WHERE series_1 IS NOT NULL
    "4-two_Partitions_timeCustom_series_2_idx" btree ("timeCustom" DESC NULLS LAST, series_2) WHERE series_2 IS NOT NULL
    "5-two_Partitions_timeCustom_series_bool_idx" btree ("timeCustom" DESC NULLS LAST, series_bool) WHERE series_bool IS NOT NULL
    "6-two_Partitions_timeCustom_device_id_idx" btree ("timeCustom" DESC NULLS LAST, device_id)
    "7-two_Partitions_timeCustom_idx" btree ("timeCustom" DESC)
Check constraints:
    "constraint_1" CHECK ("timeCustom" >= '1257892416000000000'::bigint AND "timeCustom" < '1257895008000000000'::bigint)
    "constraint_2" CHECK (_timescaledb_internal.get_partition_for_key(device_id) >= 1073741823 AND _timescaledb_internal.get_partition_for_key(device_id) < 2147483647)
Inherits: "two_Partitions"

                   Table "_timescaledb_internal._hyper_1_2_chunk"
   Column    |       Type       | Modifiers | Storage  | Stats target | Description 
-------------+------------------+-----------+----------+--------------+-------------
 timeCustom  | bigint           | not null  | plain    |              | 
 device_id   | text             | not null  | extended |              | 
 series_0    | double precision |           | plain    |              | 
 series_1    | double precision |           | plain    |              | 
 series_2    | double precision |           | plain    |              | 
 series_bool | boolean          |           | plain    |              | 
Indexes:
    "10-two_Partitions_timeCustom_series_1_idx" btree ("timeCustom" DESC NULLS LAST, series_1) WHERE series_1 IS NOT NULL
    "11-two_Partitions_timeCustom_series_2_idx" btree ("timeCustom" DESC NULLS LAST, series_2) WHERE series_2 IS NOT NULL
    "12-two_Partitions_timeCustom_series_bool_idx" btree ("timeCustom" DESC NULLS LAST, series_bool) WHERE series_bool IS NOT NULL
    "13-two_Partitions_timeCustom_device_id_idx" btree ("timeCustom" DESC NULLS LAST, device_id)
    "14-two_Partitions_timeCustom_idx" btree ("timeCustom" DESC)
    "8-two_Partitions_device_id_timeCustom_idx" btree (device_id, "timeCustom" DESC NULLS LAST) WHERE device_id IS NOT NULL
    "9-two_Partitions_timeCustom_series_0_idx" btree ("timeCustom" DESC NULLS LAST, series_0) WHERE series_0 IS NOT NULL
Check constraints:
    "constraint_2" CHECK (_timescaledb_internal.get_partition_for_key(device_id) >= 1073741823 AND _timescaledb_internal.get_partition_for_key(device_id) < 2147483647)
    "constraint_3" CHECK ("timeCustom" >= '1257897600000000000'::bigint AND "timeCustom" < '1257900192000000000'::bigint)
Inherits: "two_Partitions"

                   Table "_timescaledb_internal._hyper_1_3_chunk"
   Column    |       Type       | Modifiers | Storage  | Stats target | Description 
-------------+------------------+-----------+----------+--------------+-------------
 timeCustom  | bigint           | not null  | plain    |              | 
 device_id   | text             | not null  | extended |              | 
 series_0    | double precision |           | plain    |              | 
 series_1    | double precision |           | plain    |              | 
 series_2    | double precision |           | plain    |              | 
 series_bool | boolean          |           | plain    |              | 
Indexes:
    "15-two_Partitions_device_id_timeCustom_idx" btree (device_id, "timeCustom" DESC NULLS LAST) WHERE device_id IS NOT NULL
    "16-two_Partitions_timeCustom_series_0_idx" btree ("timeCustom" DESC NULLS LAST, series_0) WHERE series_0 IS NOT NULL
    "17-two_Partitions_timeCustom_series_1_idx" btree ("timeCustom" DESC NULLS LAST, series_1) WHERE series_1 IS NOT NULL
    "18-two_Partitions_timeCustom_series_2_idx" btree ("timeCustom" DESC NULLS LAST, series_2) WHERE series_2 IS NOT NULL
    "19-two_Partitions_timeCustom_series_bool_idx" btree ("timeCustom" DESC NULLS LAST, series_bool) WHERE series_bool IS NOT NULL
    "20-two_Partitions_timeCustom_device_id_idx" btree ("timeCustom" DESC NULLS LAST, device_id)
    "21-two_Partitions_timeCustom_idx" btree ("timeCustom" DESC)
Check constraints:
    "constraint_2" CHECK (_timescaledb_internal.get_partition_for_key(device_id) >= 1073741823 AND _timescaledb_internal.get_partition_for_key(device_id) < 2147483647)
    "constraint_4" CHECK ("timeCustom" >= '1257985728000000000'::bigint AND "timeCustom" < '1257988320000000000'::bigint)
Inherits: "two_Partitions"

                   Table "_timescaledb_internal._hyper_1_4_chunk"
   Column    |       Type       | Modifiers | Storage  | Stats target | Description 
-------------+------------------+-----------+----------+--------------+-------------
 timeCustom  | bigint           | not null  | plain    |              | 
 device_id   | text             | not null  | extended |              | 
 series_0    | double precision |           | plain    |              | 
 series_1    | double precision |           | plain    |              | 
 series_2    | double precision |           | plain    |              | 
 series_bool | boolean          |           | plain    |              | 
Indexes:
    "22-two_Partitions_device_id_timeCustom_idx" btree (device_id, "timeCustom" DESC NULLS LAST) WHERE device_id IS NOT NULL
    "23-two_Partitions_timeCustom_series_0_idx" btree ("timeCustom" DESC NULLS LAST, series_0) WHERE series_0 IS NOT NULL
    "24-two_Partitions_timeCustom_series_1_idx" btree ("timeCustom" DESC NULLS LAST, series_1) WHERE series_1 IS NOT NULL
    "25-two_Partitions_timeCustom_series_2_idx" btree ("timeCustom" DESC NULLS LAST, series_2) WHERE series_2 IS NOT NULL
    "26-two_Partitions_timeCustom_series_bool_idx" btree ("timeCustom" DESC NULLS LAST, series_bool) WHERE series_bool IS NOT NULL
    "27-two_Partitions_timeCustom_device_id_idx" btree ("timeCustom" DESC NULLS LAST, device_id)
    "28-two_Partitions_timeCustom_idx" btree ("timeCustom" DESC)
Check constraints:
    "constraint_1" CHECK ("timeCustom" >= '1257892416000000000'::bigint AND "timeCustom" < '1257895008000000000'::bigint)
    "constraint_5" CHECK (_timescaledb_internal.get_partition_for_key(device_id) >= 0 AND _timescaledb_internal.get_partition_for_key(device_id) < 1073741823)
Inherits: "two_Partitions"

SELECT * FROM _timescaledb_catalog.chunk;
 id | hypertable_id |      schema_name      |    table_name    
----+---------------+-----------------------+------------------
  1 |             1 | _timescaledb_internal | _hyper_1_1_chunk
  2 |             1 | _timescaledb_internal | _hyper_1_2_chunk
  3 |             1 | _timescaledb_internal | _hyper_1_3_chunk
  4 |             1 | _timescaledb_internal | _hyper_1_4_chunk
(4 rows)

SELECT * FROM "two_Partitions" ORDER BY "timeCustom", device_id;
     timeCustom      | device_id | series_0 | series_1 | series_2 | series_bool 
---------------------+-----------+----------+----------+----------+-------------
 1257894000000000000 | dev1      |      1.5 |        1 |        2 | t
 1257894000000000000 | dev1      |      1.5 |        2 |          | 
 1257894000000000000 | dev2      |      1.5 |        2 |          | 
 1257894000000000000 | dev2      |      1.5 |        1 |          | 
 1257894000000001000 | dev1      |      2.5 |        3 |          | 
 1257894001000000000 | dev1      |      3.5 |        4 |          | 
 1257894002000000000 | dev1      |      2.5 |        3 |          | 
 1257894002000000000 | dev1      |      5.5 |        6 |          | t
 1257894002000000000 | dev1      |      5.5 |        7 |          | f
 1257897600000000000 | dev1      |      4.5 |        5 |          | f
 1257987600000000000 | dev1      |      1.5 |        2 |          | 
 1257987600000000000 | dev1      |      1.5 |        1 |          | 
(12 rows)

SELECT * FROM ONLY "two_Partitions";
 timeCustom | device_id | series_0 | series_1 | series_2 | series_bool 
------------+-----------+----------+----------+----------+-------------
(0 rows)

CREATE TABLE error_test(time timestamp, temp float8, device text NOT NULL);
SELECT create_hypertable('error_test', 'time', 'device', 2);
 create_hypertable 
-------------------
 
(1 row)

\set QUIET off
INSERT INTO error_test VALUES ('Mon Mar 20 09:18:20.1 2017', 21.3, 'dev1');
INSERT 0 1
\set ON_ERROR_STOP 0
-- generate insert error 
INSERT INTO error_test VALUES ('Mon Mar 20 09:18:22.3 2017', 21.1, NULL);
ERROR:  null value in column "device" violates not-null constraint
\set ON_ERROR_STOP 1
INSERT INTO error_test VALUES ('Mon Mar 20 09:18:25.7 2017', 22.4, 'dev2');
INSERT 0 1
\set QUIET on
SELECT * FROM error_test;
            time            | temp | device 
----------------------------+------+--------
 Mon Mar 20 09:18:20.1 2017 | 21.3 | dev1
 Mon Mar 20 09:18:25.7 2017 | 22.4 | dev2
(2 rows)

--test character(9) partition keys since there were issues with padding causing partitioning errors
CREATE TABLE tick_character (
    symbol      character(9) NOT NULL,
    mid       REAL NOT NULL,
    spread      REAL NOT NULL,
    time        TIMESTAMPTZ       NOT NULL
);
SELECT create_hypertable ('tick_character', 'time', 'symbol', 2);
 create_hypertable 
-------------------
 
(1 row)

INSERT INTO tick_character ( symbol, mid, spread, time ) VALUES ( 'GBPJPY', 142.639000, 5.80, 'Mon Mar 20 09:18:22.3 2017') RETURNING time, symbol, mid;
              time              |  symbol   |   mid   
--------------------------------+-----------+---------
 Mon Mar 20 09:18:22.3 2017 PDT | GBPJPY    | 142.639
(1 row)

SELECT * FROM tick_character;
  symbol   |   mid   | spread |              time              
-----------+---------+--------+--------------------------------
 GBPJPY    | 142.639 |    5.8 | Mon Mar 20 09:18:22.3 2017 PDT
(1 row)

CREATE TABLE  many_partitions_test(time timestamp, temp float8, device text NOT NULL);
SELECT create_hypertable('many_partitions_test', 'time', 'device', 1000);
 create_hypertable 
-------------------
 
(1 row)

--NOTE: how much slower the first two queries are -- they are creating chunks
INSERT INTO many_partitions_test
    SELECT to_timestamp(ser), ser, ser::text FROM generate_series(1,100) ser;
INSERT INTO many_partitions_test
    SELECT to_timestamp(ser), ser, ser::text FROM generate_series(101,200) ser;
INSERT INTO many_partitions_test
    SELECT to_timestamp(ser), ser, (ser-201)::text FROM generate_series(201,300) ser;
SELECT * FROM  many_partitions_test ORDER BY time DESC LIMIT 2;
           time           | temp | device 
--------------------------+------+--------
 Wed Dec 31 16:05:00 1969 |  300 | 99
 Wed Dec 31 16:04:59 1969 |  299 | 98
(2 rows)

SELECT count(*) FROM  many_partitions_test;
 count 
-------
   300
(1 row)

CREATE TABLE  date_col_test(time date, temp float8, device text NOT NULL);
SELECT create_hypertable('date_col_test', 'time', 'device', 1000, chunk_time_interval => INTERVAL '1 Day');
 create_hypertable 
-------------------
 
(1 row)

INSERT INTO date_col_test
VALUES ('2001-02-01', 98, 'dev1'),
('2001-03-02', 98, 'dev1');
SELECT * FROM date_col_test WHERE time > '2001-01-01';
    time    | temp | device 
------------+------+--------
 02-01-2001 |   98 | dev1
 03-02-2001 |   98 | dev1
(2 rows)

